util/src/main/java/org/killbill/billing/util/cache/KillBillCacheController.java

```diff
@@ -72,7 +72,13 @@ public V get(final K key, final CacheLoaderArgument cacheLoaderArgument) {
         V value;
         try {
             if (!isKeyInCache(key)) {
-                value = computeAndCacheValue(key, cacheLoaderArgument);
+                synchronized (this) {
```

Comment:
This synchronizes all catalog operations across all tenants (maybe good enough).

If we wanted to synchronize on a per tenant level, we would need a more complicated scheme: Even though the `key` is the `tenantRecordId`, a new value of such key is passed each time, making it useless for sync.